### DISCUSSION TITLE
Lesson 2, Concept 9-10: reason for cloning?

### DISCUSSION THREAD

**Discussion by nigoshh** [OpenMined/courses#304] [Course: Foundations of Private Computation] [answered]
I'm wondering, what would happen if we don't clone the pointers? In general, is there some material on the inner workings of PySyft/Duet?

**Comment by znreza** [comment, marked as answer]
Hi @nigoshh the concept of cloning is coming from torch itself, it is not unique to PySyft/Duet. I suppose if we don't clone the pointers, then we may lose the reference of original gradients which we need for backpropagation. It's the same concept we sometime use in torch. 

Here's the [syft documentation](https://pysyft.readthedocs.io/en/latest/api/syft.core.html). Please note that it is not updated recently so may have some discrepancies with the newer releases.

**Reply by nigoshh** [reply]
Thank you @znreza! I haven't used torch very much, I should dig into that too
